File: python/9-2.py
import random

N = 10000
DIM = 5

x = [0] * DIM
r = [0] * DIM
m = [0] * DIM

print( " 回数円面積球体積4次元5次元" )
for n in range( 1, N+1 ):
    for j in range( DIM ):
        x[j] = random.random()
        r[j] = 0.0
    rr = 0.0
    for j in range(DIM):
        rr += ( x[j] - 0.5 ) * ( x[j] - 0.5 )
        r[j] = rr
    for j in range(DIM):
        if( r[j] <= 0.25 ):
            m[j] += 1
    if( ( n % 1000 ) == 0 ):
        print( f"{n:6d}", end="" )
        for j in range( 1, DIM ):
            print( f"{float(m[j])/float(n):9.4f}", end="" )
        print()

File: julia/9-2.jl
# 球体の体積計算

using Printf

const N = 10000     # 繰り返し回数
const DIM = 5       # 次元数
m = zeros(DIM)

println( " 回数\t円面積\t球体積\t4次元\t5次元")     # 見出し打出し
for n=1:N
    x = rand(5)
    r = zeros(5)
    rr = 0.0
    for j=1:DIM
        rr += ( x[j] - 0.5 )^2
        r[j] = rr
    end
    for j=2:DIM
        if r[j] < 0.25
            m[j] += 1
        end
    end
    if ( n % 1000 ) == 0
        # 解打出し
        @printf( "%6d", n )
        for j=2:DIM
            @printf( "%9.4f", m[j]/n )
        end
        println()
    end
end

